src/js/tabs/motors.js

```diff
@@ -635,27 +636,34 @@ TABS.motors.initialize = function (callback) {
             self.previousDshotBidir = FC.MOTOR_CONFIG.use_dshot_telemetry;
             self.previousFilterDynQ = FC.FILTER_CONFIG.dyn_notch_q;
             self.previousFilterDynWidth = FC.FILTER_CONFIG.dyn_notch_width_percent;
+            self.previousFilterDynCount = FC.FILTER_CONFIG.dyn_notch_count;
 
             dshotBidirElement.on("change", function () {
                 const value = $(this).prop('checked');
                 const newValue = (value !== FC.MOTOR_CONFIG.use_dshot_telemetry) ? 'On' : 'Off';
                 self.analyticsChanges['BidirectionalDshot'] = newValue;
                 FC.MOTOR_CONFIG.use_dshot_telemetry = value;
 
-                FC.FILTER_CONFIG.dyn_notch_width_percent = self.previousFilterDynWidth;
+                FC.FILTER_CONFIG.dyn_notch_count = self.previousFilterDynCount;
                 FC.FILTER_CONFIG.dyn_notch_q = self.previousFilterDynQ;
+                FC.FILTER_CONFIG.dyn_notch_width_percent = self.previousFilterDynWidth;
 
                 if (FC.FILTER_CONFIG.gyro_rpm_notch_harmonics !== 0) { // if rpm filter is active
                     if (value && !self.previousDshotBidir) {
-                        FC.FILTER_CONFIG.dyn_notch_width_percent = FILTER_DEFAULT.dyn_notch_width_percent_rpm;
+                        FC.FILTER_CONFIG.dyn_notch_count = FILTER_DEFAULT.dyn_notch_count_rpm;
                         FC.FILTER_CONFIG.dyn_notch_q = FILTER_DEFAULT.dyn_notch_q_rpm;
+                        FC.FILTER_CONFIG.dyn_notch_width_percent = FILTER_DEFAULT.dyn_notch_width_percent_rpm;
                     } else if (!value && self.previousDshotBidir) {
-                        FC.FILTER_CONFIG.dyn_notch_width_percent = FILTER_DEFAULT.dyn_notch_width_percent;
+                        FC.FILTER_CONFIG.dyn_notch_count = FILTER_DEFAULT.dyn_notch_count;
                         FC.FILTER_CONFIG.dyn_notch_q = FILTER_DEFAULT.dyn_notch_q;
+                        FC.FILTER_CONFIG.dyn_notch_width_percent = FILTER_DEFAULT.dyn_notch_width_percent;
                     }
                 }
 
-                if (FC.FILTER_CONFIG.dyn_notch_width_percent !== self.previousFilterDynWidth) {
+                const dynFilterNeedChange = (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? (FC.FILTER_CONFIG.dyn_notch_count !== self.previousFilterDynCount) :
+                (FC.FILTER_CONFIG.dyn_notch_width_percent !== self.previousFilterDynWidth);
+
```

Comment:
Indent the second line.
```suggestion
                const dynFilterNeedChange = (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? (FC.FILTER_CONFIG.dyn_notch_count !== self.previousFilterDynCount) :
                    (FC.FILTER_CONFIG.dyn_notch_width_percent !== self.previousFilterDynWidth);

```